src/libstd/collections/hash/map.rs

```diff
@@ -1009,9 +1039,7 @@ impl<K, V, S> HashMap<K, V, S>
     pub fn entry(&mut self, key: K) -> Entry<K, V> {
         // Gotta resize now.
         self.reserve(1);
-        let hash = self.make_hash(&key);
-        search_hashed(&mut self.table, hash, |q| q.eq(&key))
```

Comment:
I wasn't really sure why this method was using `search_hashed` and not `search_mut` (I think I've now made a commit with `search` instead which probably won't work).